server/routes/serviceProviderReferrals/reportingForm.ts

```diff
@@ -0,0 +1,73 @@
+import { Request } from 'express'
+import { CreateReportDateParams } from '../../services/interventionsService'
+import errorMessages from '../../utils/errorMessages'
+import CalendarDayInput from '../../utils/forms/inputs/calendarDayInput'
+import { FormValidationError } from '../../utils/formValidationError'
+
+interface ReportingDateRangeParams {
+  error: FormValidationError | null
+  reportingDateRange: CreateReportDateParams | null
+}
+
+export default class ReportingForm {
+  constructor(private readonly request: Request) {}
+
+  async data(): Promise<ReportingDateRangeParams> {
```

Comment:
We usually use `FormValidationResult` here, since it expresses that you can have precisely one of `error` or `value` being non-null.